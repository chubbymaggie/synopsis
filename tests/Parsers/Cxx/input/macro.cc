int x;

#define LONGER 12345678
#define SHORTER 1234
#define LINKINSIDE x
#define ARGS(a, b, c) x

int A = LONGER, A2 = x;
int B = SHORTER, B2 = x;
int C = LINKINSIDE, C2 = x;
int D = ARGS(1, 2, 3), D2 = x;
