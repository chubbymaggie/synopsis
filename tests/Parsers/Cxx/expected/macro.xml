<?xml version='1.0' encoding='ISO-8859-1'?>
<ast>
 <declarations>
  <instance accessibility="0" class="Synopsis.AST.Macro" file="Parsers/Cxx/input/macro.cc" language="C++" line="4" name="LONGER" text="12345678" type="macro"/>
  <instance accessibility="0" class="Synopsis.AST.Macro" file="Parsers/Cxx/input/macro.cc" language="C++" line="5" name="SHORTER" text="1234" type="macro"/>
  <instance accessibility="0" class="Synopsis.AST.Macro" file="Parsers/Cxx/input/macro.cc" language="C++" line="6" name="LINKINSIDE" text="x" type="macro"/>
  <instance accessibility="0" class="Synopsis.AST.Macro" file="Parsers/Cxx/input/macro.cc" language="C++" line="7" name="ARGS" text="x" type="macro">
   <parameters>
    a
    b
    c
   </parameters>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="1" name="x" type="global variable">
   <vtype>
    <instance class="Synopsis.Type.Base" language="C++" name="int"/>
   </vtype>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="8" name="A" type="global variable">
   <vtype/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="8" name="A2" type="global variable">
   <vtype/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="9" name="B" type="global variable">
   <vtype/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="9" name="B2" type="global variable">
   <vtype/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="10" name="C" type="global variable">
   <vtype/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="10" name="C2" type="global variable">
   <vtype/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="11" name="D" type="global variable">
   <vtype/>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Variable" constr="0" file="Parsers/Cxx/input/macro.cc" language="C++" line="11" name="D2" type="global variable">
   <vtype/>
  </instance>
 </declarations>
 <types>
  <instance class="Synopsis.Type.Declared" language="C++" name="A">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="A2">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="ARGS">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="B">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="B2">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="C">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="C2">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="D">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="D2">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="LINKINSIDE">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="LONGER">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="SHORTER">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="x">
   <declaration/>
  </instance>
 </types>
 <files>
  <file>
   Parsers/Cxx/input/macro.cc
  </file>
 </files>
</ast>
